test/e2e/parallel/space_test.go

```diff
@@ -478,6 +478,29 @@ func TestSubSpaces(t *testing.T) {
 			})
 		})
 	})
+	t.Run("we create a subSpace, with disable inheritance and do not expect roles and usernames from the parent to be inherited in NSTemplateSet", func(t *testing.T) {
+		// when
+		// we have a parentSpace
+		parentSpace, _, parentSpaceBindings := CreateSpace(t, awaitilities, testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithTierName("appstudio"))
+		// then
+		// wait until MUR and Space have been provisioned
+		VerifyResourcesProvisionedForSpace(t, awaitilities, parentSpace.Name, UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName))
+		_, err := hostAwait.WaitForMasterUserRecord(t, parentSpaceBindings.Spec.MasterUserRecord)
+		require.NoError(t, err)
+
+		// when
+		// we also have a subSpace with same tier but with disable inheritance
+		subSpace := CreateSubSpace(t, awaitilities, testspace.WithSpecParentSpace(parentSpace.Name), testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithDisableInheritance(true))
+
+		// wait until subSpace has been provisioned
+		_, subSpaceNSTemplateSet := VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name)
```

Comment:
this function expects some space roles to be present, while the subSpace has none if I'm not wrong. I think this is also the cause of the current test failure.

Reply:
yes, adding a role for  the sub sapce.